src/test/java/org/folio/entlinks/service/authority/AuthorityServiceTest.java

```diff
@@ -150,18 +151,11 @@ void shouldUpdateAuthority() {
 
     when(repository.findByIdAndDeletedFalse(id)).thenReturn(Optional.of(existed));
     when(repository.save(any(Authority.class))).thenAnswer(invocation -> invocation.getArgument(0));
-
     var updated = service.update(modified);
 
-    assertThat(updated)
-      .extracting(Authority::getId, Authority::getHeading, Authority::getHeadingType, Authority::getSource,
-        Authority::getNaturalId, Authority::getAuthoritySourceFile, Authority::getVersion, Authority::getSftHeadings,
-        Authority::getSaftHeadings, Authority::getNotes, Authority::getIdentifiers)
-      .containsExactly(modified.getId(), modified.getHeading(), modified.getHeadingType(), modified.getSource(),
-        modified.getNaturalId(), modified.getAuthoritySourceFile(), 1, modified.getSftHeadings(),
-        modified.getSaftHeadings(), modified.getNotes(), modified.getIdentifiers());
+    assertThat(updated).isEqualTo(modified);
     verify(repository).findByIdAndDeletedFalse(id);
-    verify(repository).save(existed);
+    verify(repository).save(argThat(new AuthorityMatcher(modified)));
```

Comment:
Would it work if you remove `argThat` and just pass a matcher?
You could also provide a staticMethod to return a matcher, f.e. `authorityMatch(...)`, same way as it's done in Mockito

Reply:
and I think it would be nice to have this and the other matcher for source file in a separate, some existing or new,  test utils class so that we may use it in other places as well

Reply:
@viacheslavkol If we remove an **argThat** we would not be able to make a comparison using the **ArgumentMatcher** properly